modules/clightning-plugins/teos-watchtower.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, ... }:
+
+with lib;
+let cfg = config.services.clightning.plugins.teos-watchtower; in
+{
+  # Ref.: https://github.com/talaia-labs/rust-teos/tree/master/watchtower-plugin
+  options.services.clightning.plugins.teos-watchtower = {
+    enable = mkEnableOption "TEoS watchtower (clightning plugin)";
+    package = mkOption {
+      type = types.package;
+      default = config.nix-bitcoin.pkgs.teos-watchtower-plugin;
+      defaultText = "config.nix-bitcoin.pkgs.teos-watchtower-plugin";
+      description = mdDoc "The package providing TEoS watchtower plugin binaries.";
+    };
+    port = mkOption {
+      type = types.port;
+      default = config.services.teos.port;
+      description = mdDoc "Tower API port.";
+    };
+    dataDir = mkOption {
+      type = types.path;
+      default = "${config.services.clightning.dataDir}/.watchtower";
```

Comment:
Is there a reason to make this a "hidden" directory?

Reply:
Not really actually.

I've chosen this name based on a default watchtower plugin configuration `DEFAULT_TOWERS_DATA_DIR=~/.watchtower` which should in our case match with `${config.services.clightning.dataDir}/.watchtower`:
- https://github.com/talaia-labs/rust-teos/blob/a4accedfeecf46edb9c22738ac9787b80dc8af02/watchtower-plugin/src/constants.rs#L3
- https://github.com/talaia-labs/rust-teos/blob/a4accedfeecf46edb9c22738ac9787b80dc8af02/watchtower-plugin/README.md?plain=1#L101

But we always specify the directory explicitly using the `TOWERS_DATA_DIR`, so it's probably better to not hide the directory.